### PR TITLE
Switch from using teams backend to nais api

### PR DESCRIPTION
**Pull request by thokra-nav**
NAIS-api uses a grpc client for internal communication.

This also upgrades the project to Go 1.22 because nais-api requires Go 1.22.

---

Denne krever at nais-api er på plass. Så før merging og utrulling, så må tenants utenom ci-nais og dev-nais skru av hookd reconcile i Fasit.

### CODE REVIEW

File: pkg/hookd/config/config.go [hunk 1]
```diff
@@ -32,8 +32,8 @@ type Config struct {
 	LogLinkFormatter       string        `json:"log-link-formatter"`
 	MetricsPath            string        `json:"metrics-path"`
 	ProvisionKey           string        `json:"provision-key"`
-	TeamsAPIKey            string        `json:"teams-api-key"`
-	TeamsURL               string        `json:"teams-url"`
+	NaisAPITarget          string        `json:"nais-api-target"`
```
Comment (kimtore) [inline pkg/hookd/config/config.go:35]: Rename to `nais-api-address`

File: pkg/hookd/config/config.go [hunk 2]
```diff
@@ -32,8 +32,8 @@ type Config struct {
...
+	NaisAPIInsecure        bool          `json:"nais-api-insecure"`
```
Comment (kimtore) [inline pkg/hookd/config/config.go:36]: What does "insecure" mean?
I don't mind having "insecure" in the name, but I'd prefer if the option reflected what it actually does.

File: pkg/naisapi/naisapi.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Client struct {
+	client protoapi.TeamsClient
+}
+
+func New(target string, insecureConnection bool) (*Client, error) {
```
Comment (kimtore) [inline pkg/naisapi/naisapi.go:17]: `NewClient` maybe?

**Review by kimtore — CHANGES_REQUESTED** [review]
LGTM, but some names are ambiguous, see comments.
I don't understand why the 1.22 change is needed, nais api is not a code dependency? Afaic - next time, it should be a separate PR, but for now it's fine.

**Review by kimtore — APPROVED** [review]

**Comment by thokra-nav** [comment]
The package in `./pkg/naisapi` has a dependency on [nais/api](https://github.com/nais/api), which has a dependency on `go 1.22`. 
So if I try to use `go 1.21` and add `nais/api`, I get the following message:

```
go: github.com/nais/api@v0.0.0-20240221092250-0f2590f0befc requires go >= 1.22; switching to go1.22.0
```

**Review by kimtore — CHANGES_REQUESTED** [review]
Please don't create dependencies to `nais/api` only to get the `.proto` and client code. Better to copy the proto file into this repo on-demand (with a Makefile stanza perhaps) and generate the client code locally.

**Pull request closed without merging**